corehq/apps/reports/v2/models.py

```diff
@@ -180,11 +197,43 @@ def get_context(cls):
 
 class BaseReportFilter(BaseFilter):
     name = None
+    title = None
+    endpoint_slug = None
+    ko_template_name = None
 
     def __init__(self, request, domain, context):
         self.request = request
         self.domain = domain
-        self.value = context['value']
+        self.value = context.get('value')
+        self.cache_value(self.value)
+
+    @classmethod
+    def _cache_key(cls, request, domain):
+        return "{}_{}_initial_val_{}".format(
+            request.user.username,
+            domain,
+            cls.name
+        )
+
+    @classmethod
+    def initial_value(cls, request, domain):
+        return cache.get(cls._cache_key(request, domain))
+
+    def cache_value(self, value):
+        cache.set(
+            self._cache_key(self.request, self.domain),
+            value,
+            timeout=1000 * 60 * 60 * 24 * 7,  # 7 day timeout
```

Comment:
I'm confused about what this does. When you look at a report, and add some filters, it will cache the value of the filters you entered for the next time you visit that page? But then if you don't visit after 7 days, it will get rid of those filters? 
If that's what is happening, it seems like strange UX to me, since what happens when you click on the "Explore Case Data" link will be different for different people, and different when I come back from vacation than before I left. 

Is there a reason you don't want to store the values of these filters in the URL? As I said in the last PR, doing that would also allow people to send pre-filtered reports to others.

Reply:
this is done to always load the same case type from what they were last looking at for a returning user when they refresh the page, regardless of get parameters. also it stores it according to the user & domain, but it's not meant as a way to share reports between users. it's more to always show the `case_type` a user is most interested in. I will handle the get params (or rather saved config) later on.

Reply:
Interesting. Is there a way to run searches against all case types? Or are you limited in this report to select a case type before running a search?

Reply:
we purposely limited search to one case type. it defaults to the most recently submitted case type on the very first load

Reply:
you are limited to select a case type. this was done very intentionally after talking with Dev